### PR TITLE
EditorState class added to avoid too large of files and for readablity.

### PR DESCRIPTION
**Pull request by brendajean1123**
Added EditorState class for instance in EditorTopComponent class component. 
I changed the local member variables into one instance of EditorState within EditorTopComponent class.
From:
    private boolean dirty = false;
    private boolean modified = false;
    private boolean previewDisplayed = true;
    private File currentFile;
    private DbxState dbxState;

To: 
public final class EditorState {
    private boolean dirty;
    private boolean modified;
    private boolean previewDisplayed;
    private File currentFile;
    private DbxState dbxState; 
....
Plus two constructors and basic Set and Get functions.  Not sure if it should be final class?
 I am new, so I apologize in advance if it is not done correctly.
In EditorTopComponent class is added an instance of the EditorState class and a  public function to get EditorState private variable member.

### CODE REVIEW

File: OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/EditorTopComponent.java [hunk 1]
```diff
@@ -210,22 +207,26 @@ public void redoAction() {
...
+        editorState.setStateCurrentFile(currentFile);
         setDisplayName(currentFile.getName());
     }
 
     public DbxState getDbxState() {
-        return dbxState;
+        return editorState.getStateDbxState();
     }
 
     public void setDbxState(DbxState dbxState) {
```
Comment (sebbrudzinski) [inline OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/EditorTopComponent.java]: All of the above getters/setters should be removed. Only the editorState getter/setter should be exposed and any other class that wants to modify the state should go through editor state getter/setter

File: OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/EditorState.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
+/*                    
+ * Copyright (c) 2015 Sebastian Brudzinski
```
Comment (sebbrudzinski) [inline OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/EditorState.java]: 2016

**Comment by sebbrudzinski** [comment]
It seems there are conflicts on this branch. This means that it must be merged with the latest changes on the master branch.

**Comment by brendajean1123** [comment]
Hi. I spent some time with git, syncing fork, updating fork.  Let me know if I still do not have it.  I also removed the set and get functions in the EditorState class and modified EditorTopComponent class accordingly.  Hope this is right.  Thank you for your patience.

**Comment by sebbrudzinski** [comment]
@brendajean1123 GitHub says there are still conflicts on this branch and thus it cannot be built or merged. 
Concerning EditorTopComponent - there are still getters/setters for the specific fields of EditorState. The ONLY getter available in EditorTopComponent should be the getter for EditorState.

**Pull request closed without merging**